tests/integration/drivers/vector/test_astra_db_vector_store_driver.py

```diff
@@ -0,0 +1,228 @@
+from __future__ import annotations
+
+import json
+import math
+import os
+
+import pytest
+from attrs import define, field
+
+from griptape.drivers import AstraDBVectorStoreDriver, BaseEmbeddingDriver, BaseVectorStoreDriver
+from tests.mocks.mock_tokenizer import MockTokenizer
+
+TEST_COLLECTION_NAME = "gt_int_test"
+TEST_COLLECTION_NAME_METRIC = "gt_int_test_dot"
+
+
+def astrapy_available() -> bool:
+    try:
+        return True
+    except Exception:
+        return False
+
+
+def astra_db_available() -> bool:
+    return all(
+        [
+            "ASTRA_DB_APPLICATION_TOKEN" in os.environ,
+            "ASTRA_DB_API_ENDPOINT" in os.environ,
+        ]
+    )
+
+
+def _descore_entry(entry: BaseVectorStoreDriver.Entry) -> BaseVectorStoreDriver.Entry:
+    return BaseVectorStoreDriver.Entry.from_dict({k: v for k, v in entry.__dict__.items() if k != "score"})
+
+
+@define
+class ParserEmbeddingDriver(BaseEmbeddingDriver):
+    model: str = field(default="foo", kw_only=True)
+    dimensions: int = field(default=2, kw_only=True)
+    max_attempts: int = field(default=1, kw_only=True)
+    tokenizer: MockTokenizer = field(factory=lambda: MockTokenizer(model="foo bar"), kw_only=True)
+
+    def try_embed_chunk(self, chunk: str) -> list[float]:
+        vector2d: list[float]
+        try:
+            fraction = float(json.loads(chunk))
+            angle = fraction * math.pi * 2
+            vector2d = [math.cos(angle), math.sin(angle)]
+        except Exception:
+            vector2d = [0.0, 0.0]
+        return (vector2d + [0] * (self.dimensions))[: self.dimensions]
```

Comment:
Can you use `MockEmbeddingDriver` instead? Maybe adding a `mock_output` to better control the output?

Reply:
Sure, I have subsumed the above into MockEmbeddingDriver.
Note that the tests here have to be able to dictate the resulting (2d, unit) vector computed from a string, so I have re-packed the string-to-vector logic into a function that is passed as a new field to `MockEmbeddingDriver`, called `mock_output_function`.

Is that what you suggested? (no functional changes to "ordinary" use of the MockEmbeddingDriver).

Reply:
Let's rename to `mock_output` [for consistency](https://github.com/griptape-ai/griptape/blob/6b7e8e9989f022a0743ad8b3b02d8afb348f014e/tests/mocks/mock_prompt_driver.py#L33) with `MockPromptDriver` and have the default be `Factory(lambda chunk: [0, 1])`.

Then `try_embed_chunk` can just become:
```python
    def try_embed_chunk(self, chunk: str) -> list[float]:
        return self.mock_output_function(chunk)
```

Reply:
Done (slightly different `default=` than you suggest, see comment below)

Reply:
`alternatively, I guess default=Factory(lambda: lambda chunk: [0, 1]) would work as well `

Let's actually go with this. While I think you're correct that the innermost lambda is not a dangerous mutable thing, we use this pattern in [other](https://github.com/griptape-ai/griptape/blob/d2e14b9b247010e04a72f13e8dd0b6a941015d74/griptape/tasks/base_task.py#L36-L39) [places](https://github.com/griptape-ai/griptape/blob/d2e14b9b247010e04a72f13e8dd0b6a941015d74/griptape/engines/rag/modules/retrieval/vector_store_retrieval_rag_module.py#L22-L24) throughout the framework.